src/pages/lessons/[slug].tsx

```diff
@@ -176,23 +195,19 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
             ),
           )
         }
-
-        console.debug('loading video with auth')
-        const loadedLesson = await loadLesson(initialLesson.slug)
-        console.debug('authed video loaded', {video: loadedLesson})
-
-        return {...initialLesson, ...loadedLesson}
+        // We omit the loading state here by returning lesson
+        // object that has already been loaded by videoService.
+        // TODO: Refactor/simplify the lesson machine to avoid this step entirely.
+        return lesson
```

Comment:
I didn't want to get into refactoring what's now a lesson machine, so instead I opted to just give it a lesson object once the player machine finishes loading it. works fine.